src/providers/walletconnectNotify.ts

```diff
@@ -0,0 +1,161 @@
+import fetch from 'node-fetch';
+import { capture } from '@snapshot-labs/snapshot-sentry';
+
+const WALLETCONNECT_NOTIFY_SERVER_URL = process.env.WALLETCONNECT_NOTIFY_SERVER_URL;
+const WALLETCONNECT_PROJECT_SECRET = process.env.WALLETCONNECT_PROJECT_SECRET;
+const WALLETCONNECT_PROJECT_ID = process.env.WALLETCONNECT_PROJECT_ID;
+
+const AUTH_HEADER = {
+  Authorization: WALLETCONNECT_PROJECT_SECRET ? `Bearer ${WALLETCONNECT_PROJECT_SECRET}` : ''
+};
+
+// Rate limiting numbers:
+const MAX_ACCOUNTS_PER_REQUEST = 500;
+const PER_SECOND_RATE_LIMIT = 2;
+const WAIT_ERROR_MARGIN = 0.25;
+
+// Rate limiting logic:
+async function wait(seconds: number) {
+  return new Promise<void>(resolve => {
+    setTimeout(resolve, seconds * 1_000);
+  });
+}
+
+// Match Snapshot event names to notification types
+// That should be defined in the wc-notify-config.json
+function getNotificationType(event) {
+  if (event.includes('proposal/')) {
+    return 'ed2fd071-65e1-440d-95c5-7d58884eae43';
+  } else {
+    return null;
+  }
+}
+
+// Generate a notification body per the event
+function getNotificationBody(event, space) {
+  switch (event) {
+    case 'proposal/create':
+      return `A new proposal has been created for ${space.name}`;
+    case 'proposal/end':
+      return `A proposal has closed for ${space.name}`;
```

Comment:
I talked to team about this, better we should be jusing just 'proposal/start' instead of created/end